frontend/public/components/utils/resource-log.jsx

```diff
@@ -37,188 +32,165 @@ const LogControls = ({status, toggleStreaming, dropdown, onDownload}) => {
   </div>;
 };
 
+// Resource agnostic log component
 export class ResourceLog extends SafetyFirst {
   constructor(props) {
     super(props);
-
-    this._buffer = new LineBuffer(this.props.bufferSize);
-    this._loadStarted = this._loadStarted.bind(this);
-    this._processData = this._processData.bind(this);
-    this._retry = _.throttle(this._retry, 5000);
-    this._toggleStreaming = this._toggleStreaming.bind(this);
-    this._updateStatus = this._updateStatus.bind(this);
-    this._download = this._download.bind(this);
-
+    this._buffer = [];
     this.state = {
-      touched: Date.now(),
-      status: 'loading',
+      lines: [],
+      linesBehind: 0,
+      error: false,
+      status: 'loading'
     };
   }
 
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const newState = {};
-
-    if(nextProps.eof){
-      newState.status = prevState.status === 'paused' ? 'paused' : 'eof';
+  // Download currently displayed log content
+  _download () {
+    const blob = new Blob([this._buffer.join('')], {type: 'text/plain;charset=utf-8'});
+    let filename = this.props.resourceName;
+    if (this.props.containerName) {
+      filename = `${filename}-${this.props.containerName}`;
     }
-    return newState;
-  }
-
-  componentDidMount() {
-    super.componentDidMount();
-    this._beginStreaming();
+    saveAs(blob, `${filename}.log`);
   }
 
-  componentDidUpdate(prevProps){
-    // If container changed, restart streaming
-    if (this.props.containerName && this.props.containerName !== prevProps.containerName) {
-      this._restartStreaming();
-    }
+  // Handler for websocket onclose event
+  _onClose(){
+    this.setState({ status: 'eof'});
   }
 
-  componentWillUnmount() {
-    super.componentWillUnmount();
-    this._endStreaming();
+  // Handler for websocket onerror event
+  _onError(){
```

Comment:
Still need to work out basic error handling. Not sure what the best way to handle this is. Do we have toast notifications or banners of some sort we could use here?